setup.py

```diff
@@ -162,7 +164,8 @@ def run(self):
 with open('README.rst', 'r') as readme_file:
     longdesc = readme_file.read()
 
-setup(
+
+setup_params = dict(
```

Comment:
Because the longdesc and requirements are read from files, we should still move all of those outside of the toplevel as well.  So line 134 on can move into main() (or helper functions of main() if you prefer that).

Reply:
@abadger i've started doing this and it turned out into refactoring a bit..